### PR TITLE
[kotlin] back ticks escaping of keywords and more

### LINKED ISSUE #8687: [BUG] [kotlin-spring] generator doesn't create proper inheritance hierarchy
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

The hierarchy in chase of inheritance is not created properly for kotlin language. after I fixed the mustache files the generated code is compiling.

annotations are not allowed nor necessary in the interface:

image

the interface is called, which is wrong:

image

##### openapi-generator version

5.0.1 

##### OpenAPI declaration file content or url

```yaml
 BaseFileManagerObject:
      type: object
      required:
        - id
        - name
        - self
        - objectType
      properties:
        self:
          type: string
        name:
          type: string
        parent:
          type: string
        id:
          type: string
          format: uuid
        objectType:
          type: string
      discriminator:
        propertyName: objectType
        mapping:
          FolderObject: '#/components/schemas/FolderObject'
          FileObject: '#/components/schemas/FileObject'
    FolderObject:
      type: object
      allOf:
        - $ref: '#/components/schemas/BaseFileManagerObject'
        - type: object
      required:
        - files
        - updatedAt
      properties:
        files:
          type: string
        updatedAt:
          type: string
          format: date-time
    FileObject:
      allOf:
        - $ref: '#/components/schemas/BaseFileManagerObject'
        - type: object
      required:
        - contentType
        - downloadUrl
        - scanStatus
        - size
        - updatedAt
      type: object
      properties:
        contentType:
          type: string
        size:
          type: integer
          format: int32
        downloadUrl:
          type: string
        scanStatus:
          type: string
        updatedAt:
          type: string
          format: date-time
```

##### Generation Details

I'm using the gradle plugin. this is the config:
```kotlin
tasks.named<GenerateTask>("openApiGenerate") {
  setProperty("generatorName", "kotlin-spring")
  setProperty("inputSpec", "$rootDir/swagger/swagger.yaml")
  setProperty("outputDir", "$buildDir/generated/kotlin")
//  setProperty("templateDir", "$rootDir/swagger/template/")
  setProperty("apiPackage", "io.tenera.filemanager.generated.api")
  setProperty("modelPackage", "io.tenera.filemanager.generated.model")
  setProperty("skipValidateSpec", true)
  setProperty("configOptions", mapOf(
    Pair("tile", "File Manager Api"),
    Pair("dateLibrary", "java8"),
    Pair("useBeanValidation", "true"),
    Pair("hideGenerationTimestamp", "false"),
    Pair("swaggerAnnotations", "true"),
    Pair("useTags", "true"),
    Pair("delegatePattern", "false"),
    Pair("interfaceOnly", "false"),
    Pair("serviceInterface", "true"),
    Pair("serviceImplementation", "false"),
    Pair("reactive", "false")
  ))
  setProperty("globalProperties", mapOf(
    Pair("apis", ""), //no value or comma-separated
    Pair("apiTests", "false"),
    Pair("models", ""),
    Pair("modelTests", "false"),
    Pair("invoker", "false")
  ))
}
```
##### Steps to reproduce

just run the generator for kotlin-spring with any spec with inheritance.

##### Suggest a fix

the following are my version of the templates:

dataClass.mustache (removed () in ` {{{parent}}}{{/parent}}` )
```
/**
* {{{description}}}
{{#vars}}
    * @param {{name}} {{{description}}}
{{/vars}}
*/{{#discriminator}}
    {{>typeInfoAnnotation}}{{/discriminator}}
{{#discriminator}}interface {{classname}}{{/discriminator}}{{^discriminator}}{{#hasVars}}data {{/hasVars}}class {{classname}}(
{{#requiredVars}}
    {{>dataClassReqVar}}{{^-last}},
    {{/-last}}{{/requiredVars}}{{#hasRequired}}{{#hasOptional}},
{{/hasOptional}}{{/hasRequired}}{{#optionalVars}}{{>dataClassOptVar}}{{^-last}},
{{/-last}}{{/optionalVars}}
) {{/discriminator}}{{#parent}}: {{{parent}}}{{/parent}}{
{{#discriminator}}
    {{#requiredVars}}
        {{>interfaceReqVar}}
    {{/requiredVars}}
    {{#optionalVars}}
        {{>interfaceOptVar}}
    {{/optionalVars}}
{{/discriminator}}
{{#hasEnums}}{{#vars}}{{#isEnum}}
    /**
    * {{{description}}}
    * Values: {{#allowableValues}}{{#enumVars}}{{&name}}{{^-last}},{{/-last}}{{/enumVars}}{{/allowableValues}}
    */
    enum class {{nameInCamelCase}}(val value: {{#isContainer}}{{#items}}{{{dataType}}}{{/items}}{{/isContainer}}{{^isContainer}}{{{dataType}}}{{/isContainer}}) {
    {{#allowableValues}}{{#enumVars}}
        @JsonProperty({{{value}}}) {{{name}}}({{{value}}}){{^-last}},{{/-last}}{{#-last}};{{/-last}}
    {{/enumVars}}{{/allowableValues}}
    }
{{/isEnum}}{{/vars}}{{/hasEnums}}
}
```

interfaceReqVar.mustache (added ^discriminator)

```
{{^discriminator}}
    {{#swaggerAnnotations}}
        @ApiModelProperty({{#example}}example = "{{{example}}}", {{/example}}{{#required}}required = {{required}}, {{/required}}{{#isReadOnly}}readOnly = {{{isReadOnly}}}, {{/isReadOnly}}value = "{{{description}}}"){{/swaggerAnnotations}}
{{/discriminator}}
{{>modelMutable}} {{{name}}}: {{#isEnum}}{{classname}}.{{nameInCamelCase}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}}
```

interfaceOptVar.mustache

```
{{^discriminator}}
    {{#swaggerAnnotations}}
        @ApiModelProperty({{#example}}example = "{{{example}}}", {{/example}}{{#required}}required = {{required}}, {{/required}}{{#isReadOnly}}readOnly = {{{isReadOnly}}}, {{/isReadOnly}}value = "{{{description}}}"){{/swaggerAnnotations}}
{{/discriminator}}
{{>modelMutable}} {{{name}}}: {{#isEnum}}{{classname}}.{{nameInCamelCase}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}}? {{^discriminator}}= {{#defaultvalue}}{{defaultvalue}}{{/defaultvalue}}{{^defaultvalue}}null{{/defaultvalue}}{{/discriminator}}
```

### LINKED ISSUE #11148: [BUG][KOTLIN] When using fields with reserved kotlin names and type enum, generated code contains "&#x60;" instead of backtick
##### Description
When generating kotlin code using "kotlin-spring" generator, and one of the fields in API specs is string with enum type:
```yaml
Issue:
  type: object
  properties:
    operator:
      type: string
      enum: [ One, Two ]
```

generated code contains backtick "\`" encoded as HTML entity: `&#x60;`: 
```
@field:JsonProperty("operator") val `operator`: Issue.&#x60;Operator&#x60;? = null,
```
Obviously, compiler has issue with that :)
##### openapi-generator version
5.3.0

Not regresion, I have checked 5.2.0 and 4.3.1 and all suffer from same issue. But 5.3.0 has expanded keyword list and `operator` and `value` (which we use) are escaped now.
For basic `fun` keyword it fails the same way in all versions.
##### OpenAPI declaration file content or url
```yaml
openapi: 3.0.3
info:
  title: Kotlin Issue
  version: "1.0"
paths:
  /issue:
    get:
      tags:
      - Issue
      responses:
        '200':
          description: Issue
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Issue'
components:
  schemas:
    Issue:
      type: object
      properties:
        ok:
          type: string
          enum: [ One, Two ]
        operator:
          type: string
          enum: [ One, Two ]
        fun:
          type: string
          enum: [ One, Two ]
        value:
          type: string
          enum: [ One, Two ]
```
##### Command line used for generation
gradle plugin with options:
```kotlin
    configOptions.putAll(mapOf(
        "library" to "spring-boot",
        "serializationLibrary" to "jackson",
        "interfaceOnly" to "true",
        "delegatePattern" to "true",
        "useBeanValidation" to "true",
        "sourceFolder" to "src/main/kotlin",
        "enumPropertyNaming" to "original"
    ))

```
##### Steps to reproduce
generate code and open `Issue.kt`, it will be like: 
```kotlin
data class Issue(
    @field:JsonProperty("ok") val ok: Issue.Ok? = null,
    @field:JsonProperty("operator") val `operator`: Issue.&#x60;Operator&#x60;? = null,
/* snip */
) {
enum class &#x60;Operator&#x60;(val value: kotlin.String) {
        @JsonProperty("One") One("One"),
        @JsonProperty("Two") Two("Two");
    }
}
```

### PR DESCRIPTION
**Pull request by grzegorz-moto**
This fixes #11148 where Kotlin keywords are badly escaped
This also fixes #8687 by removing `()` from base class name
Based on info from #4537 and #8687

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
cc @jimschubert (2017/09) ❤️, @dr4ke616 (2018/08) @karismann (2019/03) @Zomzog (2019/04) @andrewemery (2019/10) @4brunu (2019/11) @yutaka0m (2020/03)

File: modules/openapi-generator/src/main/resources/kotlin-spring/dataClassOptVar.mustache [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {{#useBeanValidation}}{{>beanValidation}}{{>beanValidationModel}}{{/useBeanValidation}}{{#swaggerAnnotations}}
     @ApiModelProperty({{#example}}example = "{{{.}}}", {{/example}}{{#isReadOnly}}readOnly = {{{isReadOnly}}}, {{/isReadOnly}}value = "{{{description}}}"){{/swaggerAnnotations}}{{#deprecated}}
     @Deprecated(message = ""){{/deprecated}}
-    @field:JsonProperty("{{{baseName}}}"){{#isInherited}} override{{/isInherited}} {{>modelMutable}} {{{name}}}: {{#isEnum}}{{#isArray}}{{baseType}}<{{/isArray}}{{classname}}.{{nameInCamelCase}}{{#isArray}}>{{/isArray}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}}? = {{{defaultValue}}}{{^defaultValue}}null{{/defaultValue}}
\ No newline at end of file
+    @field:JsonProperty("{{{baseName}}}"){{#isInherited}} override{{/isInherited}} {{>modelMutable}} {{{name}}}: {{#isEnum}}{{#isArray}}{{baseType}}<{{/isArray}}{{classname}}.{{{nameInCamelCase}}}{{#isArray}}>{{/isArray}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}}? = {{{defaultValue}}}{{^defaultValue}}null{{/defaultValue}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/kotlin-spring/dataClassOptVar.mustache:4]: @grzegorz-moto thanks for the PR. I think you will need to remove EOF (which is likely added automatically by your text editor)

Same for https://github.com/OpenAPITools/openapi-generator/pull/11166/files#diff-f2fe24f5af2d6f356946c5a9aeff0eedc692f07aa12d6dd8658e4c1c9bff9c38R3

Comment (grzegorz-moto) [reply modules/openapi-generator/src/main/resources/kotlin-spring/dataClassOptVar.mustache]: Done. Thanks

**Review by 4brunu — APPROVED** [review]
Looks good to me 👍

**Pull request merged**